### ISSUE TITLE
File based rules - "posted_by" left blank

### ISSUE THREAD

**Issue by t3chn0m4g3** [PaperMtn/slack-watchman#30]
When posting files I noticed that `posted_by` is left blank:
```
{"localtime": "2020-09-19 23:40:05,824", "level": "NOTIFY", "source": "Slack Watchman", "workspace": "DevCorner", "scope": "files", "detection.type": "Executable Files", "severity": "30", "detection": {"created": 1600372359, "file_id": "F01B9QV2EKT", "file_type": "binary", "mimetype": "application/octet-stream", "name": "balenaEtcher-1.5.102.dmg", "permalink": "https://slackoverloadgroup.slack.com/files/U019SN08GQP/F01B9QV2EKT/balenaetcher-1.5.102.dmg", "posted_by": "", "preview": null, "timestamp": "2020-09-16 19:27:39"}}
```
I am not sure if that is intentional since all file based rules seem to behave the same way. The username should be catched as with other detection rules.

**Comment by PaperMtn** [comment]
This seems to be a bug of the search.file method of the API: https://api.slack.com/methods/search.files, even in the example it is blank. Perhaps it is deprecated.

You can get the user ID by the look of it, and as users are enumerated and stored in memory at runtime, a username is probably retrievable that way. I'll have a look at that.

**Comment by PaperMtn** [comment]
This is fixed in the latest version (3.0.6), the API is used to get the user via the user ID

**Comment by t3chn0m4g3** [comment]
Thanks, works perfectly fine. Also thanks for updating the `type` field 😃
